plugins/inputs/opcua_listener/sample.conf

```diff
@@ -0,0 +1,116 @@
+# Retrieve data from OPCUA devices
+[[inputs.opcua]]
+  ## Metric name
+  # name = "opcua"
+  #
+  ## OPC UA Endpoint URL
+  # endpoint = "opc.tcp://localhost:4840"
+  #
+  ## Maximum time allowed to establish a connect to the endpoint.
+  # connect_timeout = "10s"
+  #
+  ## Maximum time allowed for a request over the established connection.
+  # request_timeout = "5s"
+  #
+  ## The interval at which the server should at least update its monitored items
+  # subscription_interval = "100ms"
+  #
+  ## Security policy, one of "None", "Basic128Rsa15", "Basic256",
+  ## "Basic256Sha256", or "auto"
+  # security_policy = "auto"
+  #
+  ## Security mode, one of "None", "Sign", "SignAndEncrypt", or "auto"
+  # security_mode = "auto"
+  #
+  ## Path to cert.pem. Required when security mode or policy isn't "None".
+  ## If cert path is not supplied, self-signed cert and key will be generated.
+  # certificate = "/etc/telegraf/cert.pem"
+  #
+  ## Path to private key.pem. Required when security mode or policy isn't "None".
+  ## If key path is not supplied, self-signed cert and key will be generated.
+  # private_key = "/etc/telegraf/key.pem"
+  #
+  ## Authentication Method, one of "Certificate", "UserName", or "Anonymous".  To
+  ## authenticate using a specific ID, select 'Certificate' or 'UserName'
+  # auth_method = "Anonymous"
+  #
+  ## Username. Required for auth_method = "UserName"
+  # username = ""
+  #
+  ## Password. Required for auth_method = "UserName"
+  # password = ""
+  #
+  ## Option to select the metric timestamp to use. Valid options are:
+  ##     "gather" -- uses the time of receiving the data in telegraf
+  ##     "server" -- uses the timestamp provided by the server
+  ##     "source" -- uses the timestamp provided by the source
+  # timestamp = "gather"
+  #
+  ## Node ID configuration
+  ## name              - field name to use in the output
+  ## namespace         - OPC UA namespace of the node (integer value 0 thru 3)
+  ## identifier_type   - OPC UA ID type (s=string, i=numeric, g=guid, b=opaque)
+  ## identifier        - OPC UA ID (tag as shown in opcua browser)
+  ## tags              - extra tags to be added to the output metric (optional); deprecated in 1.25.0; use default_tags
```

Comment:
It can't be deprecated if this is a brand new plugin, right?

Reply:
Yeah, I was wondering whether to add it or not. On one hand it immediately becomes deprecated, on the other, it makes the config for `opcua_listener` compatible with `opcua`. Although, under the covers it will be compatible, just not documented.

Reply:
I would not document it.